system/src/system_network_wifi.h

```diff
@@ -184,6 +184,8 @@ class WiFiNetworkInterface : public ManagedIPNetworkInterface<WLanConfig, WiFiNe
 
     bool clear_credentials() override
     {
+        system_notify_event(network_credentials, network_credentials_cleared);
```

Comment:
I feel I would like to make this conditional on the result of `wlan_clear_credentials`.  We should also check if other Radio interfaces need a similar change.  It might be better to add it to the `network_clear_credentials()` function so that it is ubiquitous?